provider/resource_harbor_webhook.go

```diff
@@ -0,0 +1,239 @@
+package provider
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/liatrio/terraform-provider-harbor/harbor"
+)
+
+func resourceWebhook() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceWebhookCreate,
+		Read:   resourceWebhookRead,
+		Update: resourceWebhookUpdate,
+		Delete: resourceWebhookDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:         schema.TypeString,
+				Description:  "ID of the project the webhook corresponds to, in the form '/projects/${ID_NUMBER}'",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^/projects/[0-9]+$`), "validation error: project_id should be of the form '/projects/${ID_NUMBER}'"),
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "Display name of the webhook.",
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Contains a description of the webhook.",
+				Optional:    true,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "When true, webhooks are enabled.",
+				Optional:    true,
+				Default:     true,
+			},
+			"event_types": {
+				Type:        schema.TypeSet,
+				Description: "The list of events that will cause this webhook to trigger.",
+				Required:    true,
+				MinItems:    1,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice(
+						[]string{
+							"DELETE_ARTIFACT",
+							"PULL_ARTIFACT",
+							"PUSH_ARTIFACT",
+							"DELETE_CHART",
+							"DOWNLOAD_CHART",
+							"UPLOAD_CHART",
+							"QUOTA_EXCEED",
+							"QUOTA_WARNING",
+							"REPLICATION",
+							"SCANNING_FAILED",
+							"SCANNING_COMPLETED",
+							"TAG_RETENTION",
+						},
+						false,
+					),
+				},
+			},
+			"target": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:         schema.TypeString,
+							Description:  "The type of notification to send.",
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"http", "slack"}, false),
+						},
+						"auth_header": {
+							Type:        schema.TypeString,
+							Description: "The webhook auth header.",
+							Optional:    true,
+						},
+						"skip_cert_verify": {
+							Type:        schema.TypeBool,
+							Description: "If true, skips tls certificate verification of webhook target.",
+							Optional:    true,
+							Default:     true,
```

Comment:
`true` does not seem like a good default for this

Reply:
True is the default selection in the UI, which is where that decision came from. Happy to switch that around though.

Reply:
Updated, default is now `false`

Reply:
perhaps we know better than their UI devs :smile: